pymc3/tests/test_sampling.py

```diff
@@ -281,19 +282,21 @@ def test_normal_vector(self, caplog):
         with model:
             # test list input
             ppc0 = pm.sample_posterior_predictive([model.test_point], samples=10)
-            ppc = pm.sample_posterior_predictive(trace, samples=10, var_names=[])
+            ppc = pm.sample_posterior_predictive(trace, samples=12, var_names=[])
```

Comment:
This is necessary for now because I rely on the trace argument to have the right length for sampling.
Since the trace is always a multi-trace, I demand that the `samples` argument be a multiple of the number of chains in the `MultiTrace`. If I knew how to take a `MultiTrace` and smash all of the component chains together into a single chain, this would not be necessary.
If you know how to do that, that would be great.

Reply:
The `get_values` method combines all chains together for a single variable. That's why `{rv_name: trace.get_values(rv_name) for rv_name in trace.varnames}` would give you a single dictionary, the keys would be the variable names and the values would be all the samples from every chain in the trace. Wouldn't this dictionary be tracelike?

Reply:
> The `get_values` method combines all chains together for a single variable. That's why `{rv_name: trace.get_values(rv_name) for rv_name in trace.varnames}` would give you a single dictionary, the keys would be the variable names and the values would be all the samples from every chain in the trace. Wouldn't this dictionary be tracelike?

Oh, excellent! I will look at that. The only remaining issue would be dealing with the absence of `trace.varnames`, and the fact that it doesn't support a `points` but that would be easy to deal with building on UserDict.  I'll see about using `get_values` and removing the limitation on allowable `samples` values.  That will be an easier canonical representation than the `MultiTrace`.

BTW, I am down to 2 tests failing, and of those 2, one would be fixed by this change and one is a multivariate issue which AFAICT simply isn't fixable by me.